src/intrinsics/node/buffer.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import invariant from "../../invariant.js";
+import { Realm } from "../../realm.js";
+import { Value, NumberValue, NativeFunctionValue, ObjectValue, StringValue } from "../../values/index.js";
+import { Set, ToInteger } from "../../methods/index.js";
+
+declare var process: any;
+
+function getBufferFromArg(realm: Realm, value: Value): Uint8Array {
+  invariant(value instanceof ObjectValue && value.$ViewedArrayBuffer);
+  let buffer = value.$ViewedArrayBuffer;
+  invariant(buffer instanceof ObjectValue && buffer.$ArrayBufferData);
+  return buffer.$ArrayBufferData;
+}
+
+export default function (realm: Realm): ObjectValue {
+  let nativeBuffer = process.binding('buffer');
+  let nativeBufferPrototype = (require('buffer'): any).Buffer.prototype;
+
+  let intrinsicName = 'process.binding("buffer")';
+  let obj = new ObjectValue(realm, realm.intrinsics.ObjectPrototype, intrinsicName);
+
+  // Buffer
+
+  let setupBufferJS = new NativeFunctionValue(realm, intrinsicName + ".setupBufferJS", "setupBufferJS", 0, (setupContext, setupArgs) => {
+    invariant(setupArgs.length === 2);
+    invariant(setupArgs[0] instanceof ObjectValue);
+    invariant(setupArgs[1] instanceof ObjectValue);
+    let [proto] = setupArgs;
```

Comment:
This is an incomplete implementation. To run this in a completely uninitialized environment, we'll have to properly implement this. It wasn't possible when I wrote it so I stopped.

It'll have to emit a new kind of side-effect that invokes this during initialization and uses a mutated value afterwards.

This is the rest of the missing implementation is here: https://github.com/nodejs/node/blob/d56a7e640f766f15980b28d15bbf02bf60975ab2/src/node_buffer.cc#L1182-L1213

All of these have corresponding definitions in C in Node.

There are three more of these incomplete setup methods inside `process`:
`_setupNextTick`
`_setupPromises`
`_setupDomainUse`